### PR TITLE
📝 Update OpenAPI client generation docs to use `@hey-api/openapi-ts`

### PR DESCRIPTION
**Pull request by jordanshatford**
Update openapi typescript client codegen docs. 

See `openapi-typescript-codegen` maintainers comment [here](https://github.com/ferdikoomen/openapi-typescript-codegen/issues/2064#issuecomment-2002593442) and [here](https://github.com/ferdikoomen/openapi-typescript-codegen/issues/1276#issuecomment-1302392146)

The [@hey-api/openapi-ts](https://github.com/hey-api/openapi-ts) repository was originally a fork, but has been detached as we plan to continue maintaining but also add improvements to the original project.

Many bugs in the original project have been fixed already in this package. We are actively tracking reported issues to the repository and will fix as quickly as possible.

Docs related:
- The command has been renamed from `openapi` to `openapi-ts` to avoid clashing.
- `useUnionTypes` was removed and is now default behaviour.
- `useOptions` is being deprecated and is now default behaviour with warning otherwise.

Disclaimer:
I am one of the people who is going to be maintaining the new package.

### CODE REVIEW

**Comment by tiangolo** [comment]
📝 Docs preview for commit 0b5533b01ddebf8ead71ba5935456886af4fdeac at: https://d05fcddf.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
📝 Docs preview for commit 565b877658180f04315a8954684ba5158d292cfb at: https://f4c44230.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
📝 Docs preview for commit 69acdcbd4bc6a94d031cb2d67183d8c7f53ffaa0 at: https://819ac387.fastapitiangolo.pages.dev

**Comment by jordanshatford** [comment]
@alejsdev is this something that can eventually be merged? If so I will make sure to keep it up to date

**Comment by tiangolo** [comment]
📝 Docs preview for commit da79f982d934f8d6a77af1bb56ce4ec4e184334c at: https://25bcd7c5.fastapitiangolo.pages.dev

**Review by codespearhead — APPROVED** [review]
For reference: the [nicolas-chaulet/openapi-typescript-codegen](https://github.com/nicolas-chaulet/openapi-typescript-codegen) repo mentioned by the original maintainer redirects to [hey-api/openapi-ts](https://github.com/hey-api/openapi-ts).

**Comment by tiangolo** [comment]
Great! Thanks @jordanshatford! :tada: 

Looking forward to your improvements in the library! :rocket: 

I'm already wishing there would be an option to expose objects with the JSON Schema, similar to the previous `--exportSchemas`, but using pure JSON Schema as is instead of a custom object shape/format. :nerd_face:

**Comment by jordanshatford** [comment]
> Great! Thanks @jordanshatford! 🎉
> 
> Looking forward to your improvements in the library! 🚀
> 
> I'm already wishing there would be an option to expose objects with the JSON Schema, similar to the previous `--exportSchemas`, but using pure JSON Schema as is instead of a custom object shape/format. 🤓

We will look into that and see! That would be a good option to provide

**Comment by mrlubos** [comment]
Thank you @tiangolo! Are/were you using the previous library? Are you able to provide more context why you'd want a JSON schema?

**Comment by tiangolo** [comment]
Thanks @mrlubos! I just did. :nerd_face:

**Comment by mrlubos** [comment]
@tiangolo just updating here as well that JSON Schemas are supported as of [v0.36.0](https://github.com/hey-api/openapi-ts/releases/tag/%40hey-api%2Fopenapi-ts%400.36.0) 🎉 Any more feedback is welcome!

**Comment by mrlubos** [comment]
@tiangolo I keep coming back to the Generate Clients tutorial page. Hey API's package keeps evolving and the tutorial is slowly getting out of sync. I don't have a good solution for this yet, but keep thinking there should be a delineation between the FastAPI -> OpenAPI step (FastAPI's domain) and OpenAPI -> TypeScript step (Hey API's domain). That way it would be easier to maintain. Sometimes I get questions about OpenAPI but it's actually something people need to handle in their FastAPI code. Maybe you're experiencing the same in reverse?

Let me know if you had any thoughts on this!

tl;dr how to handle Generate Clients tutorial getting out of sync
